test/system/surveys_test.rb

```diff
@@ -0,0 +1,51 @@
+require "application_system_test_case"
+
+class SurveysTest < ApplicationSystemTestCase
+  setup do
+    @survey = surveys(:one)
+  end
+
+  test "visiting the index" do
+    visit surveys_url
+    assert_selector "h1", text: "Surveys"
+  end
+
+  test "creating a Survey" do
+    visit surveys_url
+    click_on "New Survey"
+
+    fill_in "Comment", with: @survey.comment
+    fill_in "Difficulty", with: @survey.difficulty
+    fill_in "Interest", with: @survey.interest
+    fill_in "Realism", with: @survey.realism
+    fill_in "Submitted flag", with: @survey.submitted_flag_id
+    click_on "Create Survey"
+
+    assert_text "Survey was successfully created"
+    click_on "Back"
+  end
+
+  test "updating a Survey" do
+    visit surveys_url
+    click_on "Edit", match: :first
+
+    fill_in "Comment", with: @survey.comment
+    fill_in "Difficulty", with: @survey.difficulty
+    fill_in "Interest", with: @survey.interest
+    fill_in "Realism", with: @survey.realism
+    fill_in "Submitted flag", with: @survey.submitted_flag_id
+    click_on "Update Survey"
+
+    assert_text "Survey was successfully updated"
+    click_on "Back"
```

Comment:
Might be good to verify that the survey successfully updates in the database here.

Reply:
From what I can tell, these system tests aren't running at test time because neither Capybara or Selenium are required gems. Running the tests manually it tries to GET /surveys which doesn't exist. I think Sean was planning on having a section to edit/create survey submissions and these system tests were for that area. Not sure how he created the tests without the pages existing though.